test/hotspot/jtreg/compiler/vectorapi/TestVectorSlice.java

```diff
@@ -0,0 +1,917 @@
+                float expected;
+                if (offset < length) {
+                    expected = offset;
+                } else {
+                    expected = offset + diff - length;
+                }
+                Asserts.assertEquals(expected, dst[i][j]);
+            }
+        }
+    }
+
+    @Run(test = {"testD128", "testD256", "testD512"})
+    static void testDoubles() {
+        int maxSize = 64;
+        int diff = 70;
+        double[][] dst = new double[maxSize + 1][maxSize];
+        double[] src1 = new double[maxSize];
+        double[] src2 = new double[maxSize];
+        for (int i = 0; i < src1.length; i++) {
+            src1[i] = i;
+            src2[i] = i + diff;
+        }
+
+        int length = 2;
+        testD128(dst, src1, src2);
+        for (int i = 0; i <= length; i++) {
+            for (int j = 0; j < length; j++) {
+                int offset = i + j;
+                double expected;
+                if (offset < length) {
+                    expected = offset;
+                } else {
+                    expected = offset + diff - length;
+                }
+                Asserts.assertEquals(expected, dst[i][j]);
+            }
+        }
+
+        length = 4;
+        testD256(dst, src1, src2);
+        for (int i = 0; i <= length; i++) {
+            for (int j = 0; j < length; j++) {
+                int offset = i + j;
+                double expected;
+                if (offset < length) {
+                    expected = offset;
+                } else {
+                    expected = offset + diff - length;
+                }
+                Asserts.assertEquals(expected, dst[i][j]);
+            }
+        }
+
+        length = 8;
+        testD512(dst, src1, src2);
+        for (int i = 0; i <= length; i++) {
+            for (int j = 0; j < length; j++) {
+                int offset = i + j;
+                double expected;
+                if (offset < length) {
+                    expected = offset;
+                } else {
+                    expected = offset + diff - length;
+                }
+                Asserts.assertEquals(expected, dst[i][j]);
+            }
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.VECTOR_SLICE, "9"})
+    static void testB64(byte[][] dst, byte[] src1, byte[] src2) {
+        var species = ByteVector.SPECIES_64;
+        helper(species, dst[0], src1, src2, 0);
+        helper(species, dst[1], src1, src2, 1);
+        helper(species, dst[2], src1, src2, 2);
+        helper(species, dst[3], src1, src2, 3);
+        helper(species, dst[4], src1, src2, 4);
+        helper(species, dst[5], src1, src2, 5);
+        helper(species, dst[6], src1, src2, 6);
+        helper(species, dst[7], src1, src2, 7);
+        helper(species, dst[8], src1, src2, 8);
+    }
+
+    @Test
+    @IR(counts = {IRNode.VECTOR_SLICE, "17"})
+    static void testB128(byte[][] dst, byte[] src1, byte[] src2) {
+        var species = ByteVector.SPECIES_128;
```

Comment:
Suggest to define the species as a "`private static final`" field of this test class. It may make the intrinsification fail if the species is not a constant to the compiler.

Reply:
This local is final and is loaded from a `static final` field so it should be equivalent to referring to `ByteVector.SPECIES_128` directly